### PR TITLE
Shadergraph doc/doc12a fixes

### PR DESCRIPTION
**Pull request by ValGrimm-U3D**
# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [N/A ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [N/A ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [N/A ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [N/A ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
I made some table formatting mistakes, broke a link or two, and overwrote some of my own changes to the ToC when solving conflicts in my PRs for the documentation for Shader Graph 12/Unity 21.2. This PR fixes those errors. 

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.shadergraph/Documentation~/TableOfContents.md [hunk 1]
```diff
@@ -230,6 +233,7 @@
             * [And](And-Node)
             * [Any](Any-Node)
             * [Branch](Branch-Node)
+            *
```
Comment (marctem) [inline com.unity.shadergraph/Documentation~/TableOfContents.md:236]: Is this empty line intentional?

File: com.unity.shadergraph/Documentation~/TableOfContents.md [hunk 2]
```diff
@@ -31,7 +31,6 @@
     * [Property Types](Property-Types)
     * [Keywords](Keywords)
     * [Data Types](Data-Types)
-    * [Precision Types](Precision-Types)
```
Comment (marctem) [inline com.unity.shadergraph/Documentation~/TableOfContents.md:34]: Did you mean to remove this?

**Review by marctem — APPROVED** [review]
A couple questions, otherwise fine, feel free to merge once the comments are addressed.

File: com.unity.shadergraph/Documentation~/TableOfContents.md (hunk 1 above)
Comment (ValGrimm-U3D) [reply com.unity.shadergraph/Documentation~/TableOfContents.md]: Fixed!

File: com.unity.shadergraph/Documentation~/TableOfContents.md (hunk 2 above)
Comment (ValGrimm-U3D) [reply com.unity.shadergraph/Documentation~/TableOfContents.md:34]: Yes. Discussed with Chris back during review, it replicated part of the Precision Modes doc and was confusing in its relationship with that.

**Pull request merged**